vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py

```diff
@@ -229,16 +229,20 @@ def test_validate_step_invalid(self):
         "marks, expected",
         [
             ({i: str(i) for i in range(0, 10, 5)}, {i: str(i) for i in range(0, 10, 5)}),
-            ({15: 15, 25: 25}, {15.0: "15", 25.0: "25"}),
-            ({"15": 15, "25": 25}, {15.0: "15", 25.0: "25"}),
+            ({15: 15, 25: 25}, {15: "15", 25: "25"}),  # all int
+            ({15.5: 15.5, 25.5: 25.5}, {15.5: "15.5", 25.5: "25.5"}),  # all floats
+            ({15.0: 15, 25.0: 25}, {15: "15", 25: "25"}),  # all floats, but convertible to int
+            ({"15": 15, "25": 25}, {15: "15", 25: "25"}),  # all string
             (None, None),
         ],
     )
     def test_valid_marks(self, marks, expected):
         range_slider = vm.RangeSlider(min=0, max=10, marks=marks)
-
         assert range_slider.marks == expected
 
+        if marks:
```

Comment:
What is this? Why does the above `assert` not suffice?

Reply:
I was surprised as well, but if you remove the added assert it doesn't detect a difference between a float and an integer value e.g. this test case should fail: `({15.0: 15, 25.0: 25}, {15.0: "15", 25.0: "25"}), `

It doesn't fail if you don't have the added assert here.

Reply:
Ah, I see 👍 This is surprising to me also! TIL `assert 15 == 15.0` passes. https://www.reddit.com/r/learnpython/comments/tsb9vo/how_can_an_int_and_a_float_be_equal/